src/category_theory/category/Quiv.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.path_category
+import category_theory.category.Cat
+
+/-!
+# The category of quivers
+
+The category of (bundled) quivers, and the free/forgetful adjunction between `Cat` and `Quiv`.
+
+-/
+
+universes v u
+
+namespace category_theory
+
+/-- Category of quivers. -/
+def Quiv := bundled quiver.{(v+1) u}
+
+namespace Quiv
+
+instance : has_coe_to_sort Quiv :=
+{ S := Type u,
+  coe := bundled.α }
+
+instance str (C : Quiv.{v u}) : quiver.{(v+1) u} C := C.str
+
+/-- Construct a bundled `Quiv` from the underlying type and the typeclass. -/
+def of (C : Type u) [quiver.{v+1} C] : Quiv.{v u} := bundled.of C
+
+instance : inhabited Quiv := ⟨Quiv.of (quiver.empty pempty)⟩
+
+/-- Category structure on `Quiv` -/
+instance category : large_category.{max v u} Quiv.{v u} :=
+{ hom := λ C D, prefunctor C D,
+  id := λ C, prefunctor.id C,
+  comp := λ C D E F G, prefunctor.comp F G,
+  id_comp' := λ C D F, by cases F; refl,
+  comp_id' := λ C D F, by cases F; refl,
+  assoc' := by intros; refl }
+
+/-- The forgetful functor from categories to quivers. -/
+@[simps]
+def forget : Cat.{v u} ⥤ Quiv.{v u} :=
+{ obj := λ C, Quiv.of C,
+  map := λ C D F,
+  { obj := λ X, F.obj X,
+    map := λ X Y f, F.map f, }, }
+
+end Quiv
+
+namespace Cat
+
+local attribute [ext] functor.ext
+
+/-- The functor sending each quiver to its path category. -/
+@[simps]
+def free : Quiv.{v u} ⥤ Cat.{(max u v) u} :=
+{ obj := λ V, Cat.of (paths V),
+  map := λ V W F,
+  { obj := λ X, F.obj X,
+    map := λ X Y f, F.map_path f,
+    map_comp' := λ X Y Z f g, F.map_path_comp f g, },
+  map_id' := λ V, begin
+    ext; dsimp,
+    { induction f with b c p e ih,
+      { refl, },
+      { dsimp,
+        erw [ih, functor.id_map, functor.id_map, prefunctor.id_map],
+        simp, }, },
+    { intros X, erw [functor.id_obj, prefunctor.id_obj], refl, },
+  end,
+  map_comp' := λ U V W F G,
+  begin
+    ext; dsimp,
+    { induction f with b c p e ih,
+      { refl, },
+      { dsimp,
+        erw [ih, functor.id_map, functor.id_map, prefunctor.id_map],
+        simp, }, },
+    { intros X, erw [functor.id_obj, prefunctor.id_obj], refl, },
+  end }
+
+end Cat
+
+namespace Quiv
+
+local attribute [ext] functor.ext
+
+/-- Any prefunctor into a category lifts to a functor from the path category. -/
+@[simps]
+def lift {V : Type u} [quiver.{v+1} V] {C : Type u} [category.{v} C]
+  (F : prefunctor V C) : paths V ⥤ C :=
+{ obj := λ X, F.obj X,
+  map := λ X Y f, compose_path (F.map_path f), }
```

Comment:
You should be able to save some repetition by rearranging this file a little, proving `lift` first and promoting it to an `equiv`? I mean, the functoriality of the free functor is induced by the adjunction (do we have some result to this effect?)
```suggestion
def lift {V : Type u} [quiver.{v+1} V] {C : Type u} [category.{v} C] : 
  prefunctor V C ≃ paths V ⥤ C :=
{ to_fun := λ F, 
  { obj := λ X, F.obj X,
    map := λ X Y f, compose_path (F.map_path f) },
  inv_fun := λ F, paths.of.comp F.to_prefunctor,
  left_inv := begin
    rintro ⟨obj, map⟩,
    dsimp [prefunctor.comp], 
    congr,
    funext,
    apply category.id_comp,
  end,
  right_inv := begin
    intro F,
    ext,
    { dsimp, simp,
      induction f with Y' Z' f g ih,
      { exact (F.map_id X).symm, },
      { dsimp, simp only [ih],
        exact (F.map_comp _ _).symm, }, },
    { intro X, refl }, 
  end }
```

Reply:
I'm not sure what repetition this is avoiding. Isn't this just moving material from the adjunction up to `lift`?

(I'd prefer `lift` itself doesn't get promoted to an equivalence, to match the API in `Mat_` and `Free`, but I'm happy to have an intermediate declaration with an equivalence if it is helpful.)

Reply:
Oh, I see, you're talking about being able to skip the `map_id'` and `map_comp'` fields of `free`. I guess I'm not sure how to do that with the API we have at the moment. :-(

Reply:
I guess we are meant to set things up so we can use [`adjunction.left_adjoint_of_equiv`](https://github.com/leanprover-community/mathlib/blob/a28012c8427642abd900b51c1e6a17c888463064/src/category_theory/adjunction/basic.lean#L347).

Reply:
Yes, you do induction on paths 4 times to show that functors out of `paths V` are equal. I'm not sure `left_adjoint_of_equiv` is what we want here since it gives a less explicit action on morphisms for the free functor. I think this is fine as is -- I might submit a PR later to golf it.

Reply:
Thanks!